### ISSUE TITLE
Add tracing support for .count() and .countDocuments() in datadog-plugin-mongodb-core

### ISSUE THREAD

**Issue by patrickshuff** [DataDog/dd-trace-js#1008]
Currently there are only a limited number of mongodb-core calls that are being dynamically patched:
https://github.com/DataDog/dd-trace-js/blob/master/packages/datadog-plugin-mongodb-core/src/index.js#L139

```
function patch (core, tracer, config) {
  this.wrap(core.Server.prototype, 'command', createWrapOperation(tracer, config))
  this.wrap(core.Server.prototype, 'insert', createWrapOperation(tracer, config, 'insert'))
  this.wrap(core.Server.prototype, 'update', createWrapOperation(tracer, config, 'update'))
  this.wrap(core.Server.prototype, 'remove', createWrapOperation(tracer, config, 'remove'))

  if (core.Cursor.prototype.next) {
    this.wrap(core.Cursor.prototype, 'next', createWrapNext(tracer, config))
  } else if (core.Cursor.prototype._next) {
    this.wrap(core.Cursor.prototype, '_next', createWrapNext(tracer, config))
  }
}
```

Please add support for the default patching/tracing of both the `.count()` and the `.countDocument()` queries as they (unfortunately) can be very expensive if there are MongoDB indexes that are missing and are worth tracing by default.

**Comment by rochdev** [comment]
Can you provide a code example of how you use `count()`? I tried reproducing but I'm getting traces for `count` commands.

**Comment by rochdev** [comment]
We're looking into instrumenting the driver at the wire protocol directly which should intercept every command sent to Mongo, so this will be supported as well at that point.

**Comment by patrickshuff** [comment]
@rochdev -- we are using mongoose-paginate-v2 which was doing some .count() calls and were causing us some pain that weren't showing up in Datadog APM.

**Comment by rochdev** [comment]
Fixed in #1159
